internals/cli/masker/writer.go

```diff
@@ -155,49 +166,48 @@ func (mw *MaskedWriter) flushBuffer() {
 //
 // This should be run in a separate goroutine.
 func (mw *MaskedWriter) Run() {
+	go mw.write()
 	masking := false
 	for {
 		select {
-		case <-time.After(mw.timeout):
-			mw.lock.Lock()
-			if len(mw.output) == 0 {
-				for _, matcher := range mw.matchers {
-					matcher.Reset()
-				}
-				mw.flushBuffer()
-			}
-			mw.lock.Unlock()
 		case output := <-mw.output:
 			for _, b := range output {
 				var err error
 				if b.masked {
 					if !masking {
```

Comment:
This can definitely use a bit of documentation. If you're not already started masking the incoming bytes from `output` channel, write the full mask string `<redacted by SecretHub>` and keep dropping the bytes marked for masking until you reach the end of the byte sequence to mask.